homeassistant/components/mqtt/cover.py

```diff
@@ -443,18 +465,18 @@ def position_message_received(msg: ReceiveMessage) -> None:
                 payload = payload_dict["position"]
 
             try:
-                percentage_payload = self.find_percentage_in_range(
-                    float(payload), COVER_PAYLOAD
+                percentage_payload = ranged_value_to_percentage(
+                    self._pos_range, float(payload)
                 )
             except ValueError:
                 _LOGGER.warning("Payload '%s' is not numeric", payload)
                 return
 
-            self._attr_current_cover_position = percentage_payload
+            self._attr_current_cover_position = min(100, max(0, percentage_payload))
             if self._config.get(CONF_STATE_TOPIC) is None:
                 self._update_state(
                     STATE_CLOSED
-                    if percentage_payload == DEFAULT_POSITION_CLOSED
+                    if percentage_payload <= self._position_closed_percentage
```

Comment:
The change was incorrect. Replaced `DEFAULT_POSITION_CLOSED` (0) with `0` as the constant represents a device default position instead of the HA percentage based value needed here.